### PR TITLE
feat: implemented StarSearch waitlist

### LINKED ISSUE #3207: Feature: Waitlist
### Current behavior

<stub>

### Suggested solution

<stub>

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Adds the StarSearch waitlist feature.

## Related Tickets & Documents

Closes #3207

## Mobile & Desktop Screenshots/Recordings

CleanShot 2024-04-23 at 17 21 05

**Already on waitlist**

CleanShot 2024-04-23 at 11 24 44

**Logged out to join waitlist**

CleanShot 2024-04-23 at 11 25 27

**Logged in to join waitlist**

CleanShot 2024-04-23 at 11 26 16

## Steps to QA

Note, only the core team can really QA this as they need to reset their user profile in the development database.

To reset being on the waitlist, you'll need to run the following:

```SQL
UPDATE
	users
SET
	is_waitlisted = FALSE
WHERE
	ID = 833231; # put your user ID here instead

SELECT
	is_waitlisted,
	id
FROM
	users
WHERE
	ID = 833231; # put your user ID here instead
```

Core team members will also need to ensure these properties are false (temporarily) in Posthog. You can ask @isabensusan or myself to change it temporarily if you do not have access.

CleanShot 2024-04-24 at 09 04 16

**Logged out (not on waitlist)**

1. Go to `/star-search`
2. You're redirected to `/star-search/waitlist`.
3. Click on the `Sign up to Join the Waitlist` button.
4. You're logged in and automatically added to the waiting list.

**Logged out (on waitlist)**

1. Go to `/star-search`
2. You're redirected to `/star-search/waitlist`.
3. Click on the `Sign up to Join the Waitlist` button.
4. You're logged in and a message says you're already on the waitlist.

**Logged in (not on waitlist)**

1. Go to `/star-search`
2. You're redirected to `/star-search/waitlist`.
3. Click on the `Join the Waitlist` button.
4. You're added to the waiting list.

**Logged in (on waitlist)**

1. Go to `/star-search`
2. You're redirected to `/star-search/waitlist`.
3. A message says you're already on the waitlist.

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

### CODE REVIEW

**Comment by nickytonline** [comment]
One thing that's missing @isabensusan is the cloudy orange background. Do you have an asset for that? I can export the two layers to get pieces of it, but I don't seem to be allowed to group them in Figma.

File: pages/star-search/waitlist.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  return (
+    <ProfileLayout>
+      <div className="flex flex-col items-center gap-4 mb-16">
+        <div className="flex gap-2 items-center">
+          <Image src="/assets/star-search-logo.svg" alt="" width={40} height={40} />
+          <h1 className="text-3xl lg:text-4xl font-bold bg-clip-text text-transparent bg-gradient-to-r from-sauced-orange to-amber-400">
+            StarSearch
+          </h1>
+        </div>
+        <p className="font-semibold text-5xl">Copilot, but for your git history</p>
```
Comment (bdougie) [inline pages/star-search/waitlist.tsx:125]:
```suggestion
        <p className="font-semibold text-5xl">Copilot, but for git history</p>
```

Seeing on screen, I think maybe we remove your to broaden the ideal use case. Thoughts @isabensusan?

File: styles/globals.css [hunk 2]
```diff
@@ -226,3 +226,20 @@ but never configured it. */
 .bg-muted {
   @apply bg-slate-200;
 }
+
+.waitlist-chatbox {
```
Comment (nickytonline) [inline styles/globals.css:230]: styling for the numbered list in the mock response from StarSearch.

File: components/organisms/TopNav/top-nav.tsx [hunk 3]
```diff
@@ -85,19 +79,17 @@ const Nav = ({ className }: { className?: string }) => {
...
-                router.pathname === "/star-search"
-              )}`}
-              href={"/star-search"}
-            >
-              <img src="/assets/star-search-logo.svg" alt="Star Search logo" className="w-5 h-5" />
-              <p className="mt-0.5">StarSearch</p>
-            </Link>
-          </li>
-        )}
+        <li className="hidden lg:inline">
```
Comment (nickytonline) [inline components/organisms/TopNav/top-nav.tsx:82]: The feature flag check for the StarSearch link is no longer required as we have a waitlist page.

**Comment by nickytonline** [comment]
Here it is with the background @isabensusan.

CleanShot 2024-04-23 at 14 09 00

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Storybook build is failing.

**Comment by nickytonline** [comment]
> Storybook build is failing.

Yeah, gotta sort that out. It can't find the waitlist background image. Gonna take a peek at it before all-hands.

**Pull request merged**